src/components/common/UserProfileEdit/validation.ts

```diff
@@ -0,0 +1,46 @@
+import { ethers } from 'ethers';
+import { TestContext } from 'yup';
+
+import { intl } from '~utils/intl';
+
+const { formatMessage } = intl({
+  'rpc.invalid':
+    'RPC returned incorrect chain id ({chainId}). Expected {expectedChainId}',
+});
+
+export const isValidURL = (url: string) => {
+  if (!url) {
+    return false;
+  }
+
+  try {
+    // Tidier than using a regex. URL Constructor will error if url string invalid.
+    // eslint-disable-next-line no-new
+    new URL(url);
+    return true;
+  } catch {
+    return false;
+  }
+};
+
+async function validateCustomRPC(rpcURL: string, expectedChainId: string) {
+  const provider = new ethers.providers.JsonRpcProvider(rpcURL);
+  try {
+    const chainId = await provider.send('net_version', []);
```

Comment:
We request the network id from the rpc url given by the user.

If it's either an invalid rpc or on the wrong chain, we return an error message. Else we validate the url successfully.